prog: TestHintsRandom timed out on syzbot

```
panic: test timed out after 10m0s
running tests:
	TestHintsRandom (10m0s)

goroutine 712 [running]:
testing.(*M).startAlarm.func1()
	/usr/local/go/src/testing/testing.go:2366 +0x385
created by time.goFunc
	/usr/local/go/src/time/sleep.go:177 +0x2d

goroutine 1 [chan receive, 9 minutes]:
testing.tRunner.func1()
	/usr/local/go/src/testing/testing.go:1650 +0x4ab
testing.tRunner(0xc0001911e0, 0xc000035c70)
	/usr/local/go/src/testing/testing.go:1695 +0x134
testing.runTests(0xc0001943c0, {0x6d42500, 0x5e, 0x5e}, {0x1?, 0x5120ee?, 0x7c5f000?})
	/usr/local/go/src/testing/testing.go:2159 +0x445
testing.(*M).Run(0xc0001ee500)
	/usr/local/go/src/testing/testing.go:2027 +0x68b
main.main()
	_testmain.go:243 +0x16c

goroutine 19 [chan receive]:
github.com/google/syzkaller/pkg/stat.newSet.func1()
	/syzkaller/gopath/src/github.com/google/syzkaller/pkg/stat/set.go:95 +0x66
created by github.com/google/syzkaller/pkg/stat.newSet in goroutine 1
	/syzkaller/gopath/src/github.com/google/syzkaller/pkg/stat/set.go:94 +0xd6

goroutine 58 [runnable]:
github.com/google/syzkaller/prog.foreachArgImpl({0x1081960, 0xc0d8354ff0}, 0x0?, 0xc00945b020, 0xc030569070)
	/syzkaller/gopath/src/github.com/google/syzkaller/prog/analysis.go:155 +0x91f
github.com/google/syzkaller/prog.foreachArgImpl({0x10819a0, 0xc0d8345780}, 0x0?, 0xc00945b020, 0xc030569070)
	/syzkaller/gopath/src/github.com/google/syzkaller/prog/analysis.go:183 +0x59e
github.com/google/syzkaller/prog.foreachArgImpl({0x10819e0, 0xc0d8354fc0}, 0x0?, 0xc00945b020, 0xc030569070)
	/syzkaller/gopath/src/github.com/google/syzkaller/prog/analysis.go:205 +0x4da
github.com/google/syzkaller/prog.foreachArgImpl({0x10819a0, 0xc0d8345720}, 0x0?, 0xc00945b020, 0xc030569070)
	/syzkaller/gopath/src/github.com/google/syzkaller/prog/analysis.go:183 +0x59e
github.com/google/syzkaller/prog.foreachArgImpl({0x10819a0, 0xc0d8345700}, 0x0?, 0xc00945b020, 0xc030569070)
	/syzkaller/gopath/src/github.com/google/syzkaller/prog/analysis.go:183 +0x59e
github.com/google/syzkaller/prog.foreachArgImpl({0x10819e0, 0xc0d8354f00}, 0x0?, 0xc00945b020, 0xc030569070)
	/syzkaller/gopath/src/github.com/google/syzkaller/prog/analysis.go:205 +0x4da
github.com/google/syzkaller/prog.foreachArgImpl({0x10819a0, 0xc0d8344660}, 0x0?, 0xc00945b020, 0xc030569070)
	/syzkaller/gopath/src/github.com/google/syzkaller/prog/analysis.go:183 +0x59e
github.com/google/syzkaller/prog.foreachArgImpl({0x10819a0, 0xc0d8344640}, 0x0?, 0xc00945b020, 0xc030569070)
	/syzkaller/gopath/src/github.com/google/syzkaller/prog/analysis.go:183 +0x59e
github.com/google/syzkaller/prog.foreachArgImpl({0x10819a0, 0xc0d8344620}, 0x0?, 0xc00945b020, 0xc030569070)
	/syzkaller/gopath/src/github.com/google/syzkaller/prog/analysis.go:183 +0x59e
github.com/google/syzkaller/prog.foreachArgImpl({0x10819a0, 0xc0d8344600}, 0x0?, 0xc00945b020, 0xc030569070)
	/syzkaller/gopath/src/github.com/google/syzkaller/prog/analysis.go:183 +0x59e
github.com/google/syzkaller/prog.foreachArgImpl({0x10819e0, 0xc0d8343c50}, 0x0?, 0xc00945b020, 0xc030569070)
	/syzkaller/gopath/src/github.com/google/syzkaller/prog/analysis.go:205 +0x4da
github.com/google/syzkaller/prog.foreachArgImpl({0x10819a0, 0xc08d6e1b60}, 0x0?, 0xc00945b020, 0xc030569070)
	/syzkaller/gopath/src/github.com/google/syzkaller/prog/analysis.go:183 +0x59e
github.com/google/syzkaller/prog.foreachArgImpl({0x10819a0, 0xc08d6e1b40}, 0x0?, 0xc00945b020, 0xc030569070)
	/syzkaller/gopath/src/github.com/google/syzkaller/prog/analysis.go:183 +0x59e
github.com/google/syzkaller/prog.foreachArgImpl({0x10819e0, 0xc0d52743c0}, 0x0?, 0xc00945b020, 0xc030569070)
	/syzkaller/gopath/src/github.com/google/syzkaller/prog/analysis.go:205 +0x4da
github.com/google/syzkaller/prog.foreachArgImpl({0x10819a0, 0xc08d6e1b20}, 0x0?, 0xc00945b020, 0xc030569070)
	/syzkaller/gopath/src/github.com/google/syzkaller/prog/analysis.go:183 +0x59e
github.com/google/syzkaller/prog.foreachArgImpl({0x10819a0, 0xc08d6e1ae0}, 0x0?, 0xc00945b020, 0xc030569070)
	/syzkaller/gopath/src/github.com/google/syzkaller/prog/analysis.go:183 +0x59e
github.com/google/syzkaller/prog.foreachArgImpl({0x1081920, 0xc0d5274300}, 0x0?, 0xc00945b020, 0xc030569070)
	/syzkaller/gopath/src/github.com/google/syzkaller/prog/analysis.go:202 +0x4ad
github.com/google/syzkaller/prog.foreachArgImpl({0x10819a0, 0xc08d6e1aa0}, 0x0?, 0xc00945b020, 0xc030569070)
	/syzkaller/gopath/src/github.com/google/syzkaller/prog/analysis.go:183 +0x59e
github.com/google/syzkaller/prog.foreachArgImpl({0x1081920, 0xc0d52742d0}, 0x0?, 0xc00945b020, 0xc030569070)
	/syzkaller/gopath/src/github.com/google/syzkaller/prog/analysis.go:202 +0x4ad
github.com/google/syzkaller/prog.foreachArgImpl({0x10819a0, 0xc08d6e1a60}, 0x0?, 0xc00945b020, 0xc030569070)
	/syzkaller/gopath/src/github.com/google/syzkaller/prog/analysis.go:183 +0x59e
github.com/google/syzkaller/prog.foreachArgImpl({0x1081920, 0xc0d5274270}, 0x0?, 0xc00945b020, 0xc030569070)
	/syzkaller/gopath/src/github.com/google/syzkaller/prog/analysis.go:202 +0x4ad
github.com/google/syzkaller/prog.foreachSubArgWithStack(...)
	/syzkaller/gopath/src/github.com/google/syzkaller/prog/analysis.go:140
github.com/google/syzkaller/prog.forEachStaleUnion(0xc0001f94a0, 0xc0d5272140, 0xc0305690c8)
	/syzkaller/gopath/src/github.com/google/syzkaller/prog/expr.go:113 +0x51
github.com/google/syzkaller/prog.(*Call).checkConditions(0xc002571180?, 0x10818e0?, 0xa0?)
	/syzkaller/gopath/src/github.com/google/syzkaller/prog/expr.go:186 +0x58
github.com/google/syzkaller/prog.(*validCtx).validateCall(0xc002571180, 0xc0d5272140)
	/syzkaller/gopath/src/github.com/google/syzkaller/prog/validation.go:85 +0x185
github.com/google/syzkaller/prog.(*Prog).validateWithOpts(0xc0d89ef780, {0x0?})
	/syzkaller/gopath/src/github.com/google/syzkaller/prog/validation.go:57 +0x11a
github.com/google/syzkaller/prog.(*Prog).validate(...)
	/syzkaller/gopath/src/github.com/google/syzkaller/prog/validation.go:30
github.com/google/syzkaller/prog.(*Prog).debugValidate(...)
	/syzkaller/gopath/src/github.com/google/syzkaller/prog/validation.go:23
github.com/google/syzkaller/prog.(*Prog).MutateWithHints.func1()
	/syzkaller/gopath/src/github.com/google/syzkaller/prog/hints.go:119 +0x6f
github.com/google/syzkaller/prog.checkDataArg(0xc0d77dbaa0?, 0xc03056bd78, 0xc03056bb38)
	/syzkaller/gopath/src/github.com/google/syzkaller/prog/hints.go:222 +0x1d7
github.com/google/syzkaller/prog.generateHints(0xc03056bd78, {0x1081960, 0xc0d77e0a50}, 0x0, 0xc03056bb38)
	/syzkaller/gopath/src/github.com/google/syzkaller/prog/hints.go:180 +0x216
github.com/google/syzkaller/prog.(*Prog).MutateWithHints.func2({0x1081960?, 0xc0d77e0a50?}, 0x0?)
	/syzkaller/gopath/src/github.com/google/syzkaller/prog/hints.go:128 +0x3c
github.com/google/syzkaller/prog.foreachArgImpl({0x1081960, 0xc0d77e0a50}, 0x7fc738e16d00?, 0xc0cf46a000, 0xc03056bc20)
	/syzkaller/gopath/src/github.com/google/syzkaller/prog/analysis.go:166 +0x2bf
github.com/google/syzkaller/prog.foreachArgImpl({0x10819a0, 0xc0d77d5c60}, 0x4?, 0xc0cf46a000, 0xc03056bc20)
	/syzkaller/gopath/src/github.com/google/syzkaller/prog/analysis.go:183 +0x59e
github.com/google/syzkaller/prog.foreachArgImpl({0x10819e0, 0xc0d77e0a20}, 0xc0840b8e10?, 0xc0cf46a000, 0xc03056bc20)
	/syzkaller/gopath/src/github.com/google/syzkaller/prog/analysis.go:205 +0x4da
github.com/google/syzkaller/prog.foreachArgImpl({0x10819a0, 0xc0d77d5c00}, 0x7fc738e15100?, 0xc0cf46a000, 0xc03056bc20)
	/syzkaller/gopath/src/github.com/google/syzkaller/prog/analysis.go:183 +0x59e
github.com/google/syzkaller/prog.foreachArgImpl({0x10819a0, 0xc0d77d5be0}, 0x4?, 0xc0cf46a000, 0xc03056bc20)
	/syzkaller/gopath/src/github.com/google/syzkaller/prog/analysis.go:183 +0x59e
github.com/google/syzkaller/prog.foreachArgImpl({0x10819e0, 0xc0d77e0900}, 0x7fc738e15100?, 0xc0cf46a000, 0xc03056bc20)
	/syzkaller/gopath/src/github.com/google/syzkaller/prog/analysis.go:205 +0x4da
github.com/google/syzkaller/prog.foreachArgImpl({0x10819a0, 0xc0d77d5a40}, 0x4?, 0xc0cf46a000, 0xc03056bc20)
	/syzkaller/gopath/src/github.com/google/syzkaller/prog/analysis.go:183 +0x59e
github.com/google/syzkaller/prog.foreachArgImpl({0x10819a0, 0xc0d77d5a20}, 0x8356e9?, 0xc0cf46a000, 0xc03056bc20)
	/syzkaller/gopath/src/github.com/google/syzkaller/prog/analysis.go:183 +0x59e
github.com/google/syzkaller/prog.foreachArgImpl({0x10819a0, 0xc0d77d5a00}, 0x8?, 0xc0cf46a000, 0xc03056bc20)
	/syzkaller/gopath/src/github.com/google/syzkaller/prog/analysis.go:183 +0x59e
github.com/google/syzkaller/prog.foreachArgImpl({0x10819a0, 0xc0d77d59e0}, 0xc0840b8e10?, 0xc0cf46a000, 0xc03056bc20)
	/syzkaller/gopath/src/github.com/google/syzkaller/prog/analysis.go:183 +0x59e
github.com/google/syzkaller/prog.foreachArgImpl({0x10819e0, 0xc0d77e04e0}, 0x7fc738e27a00?, 0xc0cf46a000, 0xc03056bc20)
	/syzkaller/gopath/src/github.com/google/syzkaller/prog/analysis.go:205 +0x4da
github.com/google/syzkaller/prog.foreachArgImpl({0x10819a0, 0xc08d6e1b60}, 0x4?, 0xc0cf46a000, 0xc03056bc20)
	/syzkaller/gopath/src/github.com/google/syzkaller/prog/analysis.go:183 +0x59e
github.com/google/syzkaller/prog.foreachArgImpl({0x10819a0, 0xc08d6e1b40}, 0x4?, 0xc0cf46a000, 0xc03056bc20)
	/syzkaller/gopath/src/github.com/google/syzkaller/prog/analysis.go:183 +0x59e
github.com/google/syzkaller/prog.foreachArgImpl({0x10819e0, 0xc0d52743c0}, 0x0?, 0xc0cf46a000, 0xc0d914dc20)
	/syzkaller/gopath/src/github.com/google/syzkaller/prog/analysis.go:205 +0x4da
github.com/google/syzkaller/prog.foreachArgImpl({0x10819a0, 0xc08d6e1b20}, 0x0?, 0xc0cf46a000, 0xc03056bc20)
	/syzkaller/gopath/src/github.com/google/syzkaller/prog/analysis.go:183 +0x59e
github.com/google/syzkaller/prog.foreachArgImpl({0x10819a0, 0xc08d6e1ae0}, 0x7fc738e18400?, 0xc0cf46a000, 0xc03056bc20)
	/syzkaller/gopath/src/github.com/google/syzkaller/prog/analysis.go:183 +0x59e
github.com/google/syzkaller/prog.foreachArgImpl({0x1081920, 0xc0d5274300}, 0x4?, 0xc0cf46a000, 0xc0d914dc20)
	/syzkaller/gopath/src/github.com/google/syzkaller/prog/analysis.go:202 +0x4ad
github.com/google/syzkaller/prog.foreachArgImpl({0x10819a0, 0xc08d6e1aa0}, 0x0?, 0xc0cf46a000, 0xc03056bc20)
	/syzkaller/gopath/src/github.com/google/syzkaller/prog/analysis.go:183 +0x59e
github.com/google/syzkaller/prog.foreachArgImpl({0x1081920, 0xc0d52742d0}, 0x0?, 0xc0cf46a000, 0xc0d914dc20)
	/syzkaller/gopath/src/github.com/google/syzkaller/prog/analysis.go:202 +0x4ad
github.com/google/syzkaller/prog.foreachArgImpl({0x10819a0, 0xc08d6e1a60}, 0x7fc6545cb060?, 0xc0cf46a000, 0xc03056bc20)
	/syzkaller/gopath/src/github.com/google/syzkaller/prog/analysis.go:183 +0x59e
github.com/google/syzkaller/prog.foreachArgImpl({0x1081920, 0xc0d5274270}, 0x5?, 0xc0cf46a000, 0xc0d914dc20)
	/syzkaller/gopath/src/github.com/google/syzkaller/prog/analysis.go:202 +0x4ad
github.com/google/syzkaller/prog.ForeachArg(0xc0d5272140, 0xc03056bc20)
	/syzkaller/gopath/src/github.com/google/syzkaller/prog/analysis.go:151 +0xeb
github.com/google/syzkaller/prog.(*Prog).MutateWithHints(0xc0a061e140, 0x4, 0xc0d914dd78, 0xe70d08)
	/syzkaller/gopath/src/github.com/google/syzkaller/prog/hints.go:123 +0x15c
github.com/google/syzkaller/prog.TestHintsRandom(0xc0004b5ba0?)
	/syzkaller/gopath/src/github.com/google/syzkaller/prog/hints_test.go:667 +0x21c
testing.tRunner(0xc0004b5ba0, 0xe709f8)
	/usr/local/go/src/testing/testing.go:1689 +0xfb
created by testing.(*T).Run in goroutine 1
	/usr/local/go/src/testing/testing.go:1742 +0x390
FAIL	github.com/google/syzkaller/prog	600.784s
```

`seed=1747828233088619214` times out quite consistently:

```
$ SYZ_SEED=1747828233088619214 go test -short -v  -run TestHintsRandom ./prog -count=10
=== RUN   TestHintsRandom
=== PAUSE TestHintsRandom
=== CONT  TestHintsRandom
    testutil.go:35: seed=1747828233088619214
--- PASS: TestHintsRandom (369.56s)
=== RUN   TestHintsRandom
=== PAUSE TestHintsRandom
=== CONT  TestHintsRandom
    testutil.go:35: seed=1747828233088619214
panic: test timed out after 10m0s
	running tests:
		TestHintsRandom (3m50s)

goroutine 306 [running]:
testing.(*M).startAlarm.func1()
	/usr/lib/google-golang/src/testing/testing.go:2504 +0x394
created by time.goFunc
	/usr/lib/google-golang/src/time/sleep.go:215 +0x2d

goroutine 1 [chan receive, 3 minutes]:
testing.tRunner.func1()
	/usr/lib/google-golang/src/testing/testing.go:1759 +0x4a6
testing.tRunner(0xc003303c00, 0xc000339c70)
	/usr/lib/google-golang/src/testing/testing.go:1804 +0x123
testing.runTests(0xc00052c318, {0x1b149e0, 0x5e, 0x5e}, {0x1b26880?, 0x7?, 0x1b250c0?})
	/usr/lib/google-golang/src/testing/testing.go:2284 +0x4b4
testing.(*M).Run(0xc0001c2c80)
	/usr/lib/google-golang/src/testing/testing.go:2149 +0x65a
main.main()
	_testmain.go:241 +0x9b

goroutine 18 [chan receive]:
github.com/google/syzkaller/pkg/stat.newSet.func1()
	/usr/local/google/src/syzkaller-experiment-removeme/syzkaller/pkg/stat/set.go:95 +0x66
created by github.com/google/syzkaller/pkg/stat.newSet in goroutine 1
	/usr/local/google/src/syzkaller-experiment-removeme/syzkaller/pkg/stat/set.go:94 +0xd6

goroutine 290 [runnable]:
github.com/google/syzkaller/prog.(*ConstArg).Size(0xc00185ca40?)
	/usr/local/google/src/syzkaller-experiment-removeme/syzkaller/prog/prog.go:116 +0x8b
github.com/google/syzkaller/prog.(*GroupArg).Size(0xc00185ca60)
	/usr/local/google/src/syzkaller-experiment-removeme/syzkaller/prog/prog.go:268 +0x1ce
github.com/google/syzkaller/prog.(*UnionArg).Size(0xc00bf69d70)
	/usr/local/google/src/syzkaller-experiment-removeme/syzkaller/prog/prog.go:319 +0x144
github.com/google/syzkaller/prog.(*GroupArg).Size(0xc00185c9e0)
	/usr/local/google/src/syzkaller-experiment-removeme/syzkaller/prog/prog.go:281 +0x2d1
github.com/google/syzkaller/prog.(*GroupArg).Size(0xc00185c9c0)
	/usr/local/google/src/syzkaller-experiment-removeme/syzkaller/prog/prog.go:268 +0x1ce
github.com/google/syzkaller/prog.(*UnionArg).Size(0xc00bf69ce0)
	/usr/local/google/src/syzkaller-experiment-removeme/syzkaller/prog/prog.go:319 +0x144
github.com/google/syzkaller/prog.(*GroupArg).Size(0xc00185c980)
	/usr/local/google/src/syzkaller-experiment-removeme/syzkaller/prog/prog.go:281 +0x2d1
github.com/google/syzkaller/prog.foreachArgImpl({0x15b37a0, 0xc00185c960}, 0x0?, 0xc035689860, 0xc00bff8518)
	/usr/local/google/src/syzkaller-experiment-removeme/syzkaller/prog/analysis.go:184 +0x58b
github.com/google/syzkaller/prog.foreachArgImpl({0x15b37e0, 0xc00bf69bf0}, 0x0?, 0xc035689860, 0xc00bff8518)
	/usr/local/google/src/syzkaller-experiment-removeme/syzkaller/prog/analysis.go:205 +0x4b4
github.com/google/syzkaller/prog.foreachArgImpl({0x15b37a0, 0xc00185c6c0}, 0x0?, 0xc035689860, 0xc00bff8518)
	/usr/local/google/src/syzkaller-experiment-removeme/syzkaller/prog/analysis.go:183 +0x578
github.com/google/syzkaller/prog.foreachArgImpl({0x15b37a0, 0xc00185c6a0}, 0x0?, 0xc035689860, 0xc00bff8518)
	/usr/local/google/src/syzkaller-experiment-removeme/syzkaller/prog/analysis.go:183 +0x578
github.com/google/syzkaller/prog.foreachArgImpl({0x15b37a0, 0xc001784da0}, 0x0?, 0xc035689860, 0xc00bff8518)
	/usr/local/google/src/syzkaller-experiment-removeme/syzkaller/prog/analysis.go:183 +0x578
github.com/google/syzkaller/prog.foreachArgImpl({0x15b37a0, 0xc001784d80}, 0x0?, 0xc035689860, 0xc00bff8518)
	/usr/local/google/src/syzkaller-experiment-removeme/syzkaller/prog/analysis.go:183 +0x578
github.com/google/syzkaller/prog.foreachArgImpl({0x15b37e0, 0xc00bf63170}, 0x0?, 0xc035689860, 0xc00bff8518)
	/usr/local/google/src/syzkaller-experiment-removeme/syzkaller/prog/analysis.go:205 +0x4b4
github.com/google/syzkaller/prog.foreachArgImpl({0x15b37a0, 0xc003656b60}, 0x0?, 0xc035689860, 0xc00bff8518)
	/usr/local/google/src/syzkaller-experiment-removeme/syzkaller/prog/analysis.go:183 +0x578
github.com/google/syzkaller/prog.foreachArgImpl({0x15b37a0, 0xc003656b40}, 0x0?, 0xc035689860, 0xc00bff8518)
	/usr/local/google/src/syzkaller-experiment-removeme/syzkaller/prog/analysis.go:183 +0x578
github.com/google/syzkaller/prog.foreachArgImpl({0x15b37e0, 0xc00bf35890}, 0x0?, 0xc035689860, 0xc00bff8518)
	/usr/local/google/src/syzkaller-experiment-removeme/syzkaller/prog/analysis.go:205 +0x4b4
github.com/google/syzkaller/prog.foreachArgImpl({0x15b37a0, 0xc003656ae0}, 0x0?, 0xc035689860, 0xc00bff8518)
	/usr/local/google/src/syzkaller-experiment-removeme/syzkaller/prog/analysis.go:183 +0x578
github.com/google/syzkaller/prog.foreachArgImpl({0x15b37a0, 0xc003656a80}, 0x0?, 0xc035689860, 0xc00bff8518)
	/usr/local/google/src/syzkaller-experiment-removeme/syzkaller/prog/analysis.go:183 +0x578
github.com/google/syzkaller/prog.foreachArgImpl({0x15b3720, 0xc00bf356b0}, 0x0?, 0xc035689860, 0xc00bff8518)
	/usr/local/google/src/syzkaller-experiment-removeme/syzkaller/prog/analysis.go:202 +0x478
github.com/google/syzkaller/prog.foreachArgImpl({0x15b37a0, 0xc003656a20}, 0x0?, 0xc035689860, 0xc00bff8518)
	/usr/local/google/src/syzkaller-experiment-removeme/syzkaller/prog/analysis.go:183 +0x578
github.com/google/syzkaller/prog.foreachArgImpl({0x15b3720, 0xc00bf35680}, 0x0?, 0xc035689860, 0xc00bff8518)
	/usr/local/google/src/syzkaller-experiment-removeme/syzkaller/prog/analysis.go:202 +0x478
github.com/google/syzkaller/prog.foreachArgImpl({0x15b37a0, 0xc0036569e0}, 0x0?, 0xc035689860, 0xc00bff8518)
	/usr/local/google/src/syzkaller-experiment-removeme/syzkaller/prog/analysis.go:183 +0x578
github.com/google/syzkaller/prog.foreachArgImpl({0x15b3720, 0xc00bf35620}, 0x0?, 0xc035689860, 0xc00bff8518)
	/usr/local/google/src/syzkaller-experiment-removeme/syzkaller/prog/analysis.go:202 +0x478
github.com/google/syzkaller/prog.foreachSubArgWithStack(...)
	/usr/local/google/src/syzkaller-experiment-removeme/syzkaller/prog/analysis.go:140
github.com/google/syzkaller/prog.forEachStaleUnion(0xc00046e000, 0xc000c8dea0, 0xc00bff8570)
	/usr/local/google/src/syzkaller-experiment-removeme/syzkaller/prog/expr.go:113 +0x51
github.com/google/syzkaller/prog.(*Call).checkConditions(0xc0031dea20?, 0x15b36e0?, 0x10?)
	/usr/local/google/src/syzkaller-experiment-removeme/syzkaller/prog/expr.go:186 +0x4f
github.com/google/syzkaller/prog.(*validCtx).validateCall(0xc0031dea20, 0xc000c8dea0)
	/usr/local/google/src/syzkaller-experiment-removeme/syzkaller/prog/validation.go:85 +0x185
github.com/google/syzkaller/prog.(*Prog).validateWithOpts(0xc001bde240, {0xff?})
	/usr/local/google/src/syzkaller-experiment-removeme/syzkaller/prog/validation.go:57 +0x11a
github.com/google/syzkaller/prog.(*Prog).validate(...)
	/usr/local/google/src/syzkaller-experiment-removeme/syzkaller/prog/validation.go:30
github.com/google/syzkaller/prog.(*Prog).debugValidate(...)
	/usr/local/google/src/syzkaller-experiment-removeme/syzkaller/prog/validation.go:23
github.com/google/syzkaller/prog.(*Prog).MutateWithHints.func1()
	/usr/local/google/src/syzkaller-experiment-removeme/syzkaller/prog/hints.go:119 +0x6f
github.com/google/syzkaller/prog.checkConstArg(0xc001c47130, 0x0, 0xc00bffbd80, 0xc00bffbb50)
	/usr/local/google/src/syzkaller-experiment-removeme/syzkaller/prog/hints.go:201 +0x2a6
github.com/google/syzkaller/prog.generateHints(0xc00bffbd80, {0x15b36e0, 0xc001c47130}, 0x0, 0xc00bffbb50)
	/usr/local/google/src/syzkaller-experiment-removeme/syzkaller/prog/hints.go:170 +0x227
github.com/google/syzkaller/prog.(*Prog).MutateWithHints.func2({0x15b36e0?, 0xc001c47130?}, 0x0?)
	/usr/local/google/src/syzkaller-experiment-removeme/syzkaller/prog/hints.go:128 +0x3c
github.com/google/syzkaller/prog.foreachArgImpl({0x15b36e0, 0xc001c47130}, 0x40033974b00?, 0xc03568a300, 0xc00bffbc30)
	/usr/local/google/src/syzkaller-experiment-removeme/syzkaller/prog/analysis.go:166 +0x2a8
github.com/google/syzkaller/prog.foreachArgImpl({0x15b37a0, 0xc00277a120}, 0x4?, 0xc03568a300, 0xc00bffbc30)
	/usr/local/google/src/syzkaller-experiment-removeme/syzkaller/prog/analysis.go:183 +0x578
github.com/google/syzkaller/prog.foreachArgImpl({0x15b37e0, 0xc00bf7d260}, 0x0?, 0xc03568a300, 0xc00bffbc30)
	/usr/local/google/src/syzkaller-experiment-removeme/syzkaller/prog/analysis.go:205 +0x4b4
github.com/google/syzkaller/prog.foreachArgImpl({0x15b37a0, 0xc00277a0e0}, 0x40033974900?, 0xc03568a300, 0xc00bffbc30)
	/usr/local/google/src/syzkaller-experiment-removeme/syzkaller/prog/analysis.go:183 +0x578
github.com/google/syzkaller/prog.foreachArgImpl({0x15b37a0, 0xc00277a0c0}, 0x4?, 0xc03568a300, 0xc00bffbc30)
	/usr/local/google/src/syzkaller-experiment-removeme/syzkaller/prog/analysis.go:183 +0x578
github.com/google/syzkaller/prog.foreachArgImpl({0x15b37e0, 0xc00bf7d1d0}, 0xc001f3a2b8?, 0xc03568a300, 0xc00bffbc30)
	/usr/local/google/src/syzkaller-experiment-removeme/syzkaller/prog/analysis.go:205 +0x4b4
github.com/google/syzkaller/prog.foreachArgImpl({0x15b37a0, 0xc00277a0a0}, 0x40002cdf500?, 0xc03568a300, 0xc00bffbc30)
	/usr/local/google/src/syzkaller-experiment-removeme/syzkaller/prog/analysis.go:183 +0x578
github.com/google/syzkaller/prog.foreachArgImpl({0x15b37a0, 0xc00277a080}, 0xb34780?, 0xc03568a300, 0xc00bffbc30)
	/usr/local/google/src/syzkaller-experiment-removeme/syzkaller/prog/analysis.go:183 +0x578
github.com/google/syzkaller/prog.foreachArgImpl({0x15b37e0, 0xc00bf7d1a0}, 0xc001f3a2b8?, 0xc03568a300, 0xc00bffbc30)
	/usr/local/google/src/syzkaller-experiment-removeme/syzkaller/prog/analysis.go:205 +0x4b4
github.com/google/syzkaller/prog.foreachArgImpl({0x15b37a0, 0xc00272f800}, 0xc035803010?, 0xc03568a300, 0xc00bffbc30)
	/usr/local/google/src/syzkaller-experiment-removeme/syzkaller/prog/analysis.go:183 +0x578
github.com/google/syzkaller/prog.foreachArgImpl({0x15b37a0, 0xc00272f7e0}, 0xb34780?, 0xc03568a300, 0xc00bffbc30)
	/usr/local/google/src/syzkaller-experiment-removeme/syzkaller/prog/analysis.go:183 +0x578
github.com/google/syzkaller/prog.foreachArgImpl({0x15b37e0, 0xc00bf7bad0}, 0x40002cdef00?, 0xc03568a300, 0xc00bffbc30)
	/usr/local/google/src/syzkaller-experiment-removeme/syzkaller/prog/analysis.go:205 +0x4b4
github.com/google/syzkaller/prog.foreachArgImpl({0x15b37a0, 0xc00272eb20}, 0x4?, 0xc03568a300, 0xc00bffbc30)
	/usr/local/google/src/syzkaller-experiment-removeme/syzkaller/prog/analysis.go:183 +0x578
github.com/google/syzkaller/prog.foreachArgImpl({0x15b37a0, 0xc00272eb00}, 0x9c8d6e?, 0xc03568a300, 0xc00bffbc30)
	/usr/local/google/src/syzkaller-experiment-removeme/syzkaller/prog/analysis.go:183 +0x578
github.com/google/syzkaller/prog.foreachArgImpl({0x15b37a0, 0xc00272eac0}, 0x2cc?, 0xc03568a300, 0xc00bffbc30)
	/usr/local/google/src/syzkaller-experiment-removeme/syzkaller/prog/analysis.go:183 +0x578
github.com/google/syzkaller/prog.foreachArgImpl({0x15b37a0, 0xc00272eaa0}, 0xc001f3a2b8?, 0xc03568a300, 0xc00bffbc30)
	/usr/local/google/src/syzkaller-experiment-removeme/syzkaller/prog/analysis.go:183 +0x578
github.com/google/syzkaller/prog.foreachArgImpl({0x15b37e0, 0xc00bf79b30}, 0x0?, 0xc03568a300, 0xc00bffbc30)
	/usr/local/google/src/syzkaller-experiment-removeme/syzkaller/prog/analysis.go:205 +0x4b4
github.com/google/syzkaller/prog.foreachArgImpl({0x15b37a0, 0xc003656b60}, 0xb34780?, 0xc03568a300, 0xc00bffbc30)
	/usr/local/google/src/syzkaller-experiment-removeme/syzkaller/prog/analysis.go:183 +0x578
github.com/google/syzkaller/prog.foreachArgImpl({0x15b37a0, 0xc003656b40}, 0x4?, 0xc03568a300, 0xc00bffbc30)
	/usr/local/google/src/syzkaller-experiment-removeme/syzkaller/prog/analysis.go:183 +0x578
github.com/google/syzkaller/prog.foreachArgImpl({0x15b37e0, 0xc00bf35890}, 0x0?, 0xc03568a300, 0xc00bffbc30)
	/usr/local/google/src/syzkaller-experiment-removeme/syzkaller/prog/analysis.go:205 +0x4b4
github.com/google/syzkaller/prog.foreachArgImpl({0x15b37a0, 0xc003656ae0}, 0x0?, 0xc03568a300, 0xc00bffbc30)
	/usr/local/google/src/syzkaller-experiment-removeme/syzkaller/prog/analysis.go:183 +0x578
github.com/google/syzkaller/prog.foreachArgImpl({0x15b37a0, 0xc003656a80}, 0x4000bfa2d00?, 0xc03568a300, 0xc00bffbc30)
	/usr/local/google/src/syzkaller-experiment-removeme/syzkaller/prog/analysis.go:183 +0x578
github.com/google/syzkaller/prog.foreachArgImpl({0x15b3720, 0xc00bf356b0}, 0x4?, 0xc03568a300, 0xc00bffbc30)
	/usr/local/google/src/syzkaller-experiment-removeme/syzkaller/prog/analysis.go:202 +0x478
github.com/google/syzkaller/prog.foreachArgImpl({0x15b37a0, 0xc003656a20}, 0x0?, 0xc03568a300, 0xc00bffbc30)
	/usr/local/google/src/syzkaller-experiment-removeme/syzkaller/prog/analysis.go:183 +0x578
github.com/google/syzkaller/prog.foreachArgImpl({0x15b3720, 0xc00bf35680}, 0x0?, 0xc03568a300, 0xc00bffbc30)
	/usr/local/google/src/syzkaller-experiment-removeme/syzkaller/prog/analysis.go:202 +0x478
github.com/google/syzkaller/prog.foreachArgImpl({0x15b37a0, 0xc0036569e0}, 0xc00bffb928?, 0xc03568a300, 0xc00bffbc30)
	/usr/local/google/src/syzkaller-experiment-removeme/syzkaller/prog/analysis.go:183 +0x578
github.com/google/syzkaller/prog.foreachArgImpl({0x15b3720, 0xc00bf35620}, 0x5?, 0xc03568a300, 0xc00bffbc30)
	/usr/local/google/src/syzkaller-experiment-removeme/syzkaller/prog/analysis.go:202 +0x478
github.com/google/syzkaller/prog.ForeachArg(0xc000c8dea0, 0xc00bffbc30)
	/usr/local/google/src/syzkaller-experiment-removeme/syzkaller/prog/analysis.go:151 +0xed
github.com/google/syzkaller/prog.(*Prog).MutateWithHints(0xc025c409c0, 0x4, 0xc00bffbd80, 0x1500b30)
	/usr/local/google/src/syzkaller-experiment-removeme/syzkaller/prog/hints.go:123 +0x148
github.com/google/syzkaller/prog.TestHintsRandom(0xc003303dc0?)
	/usr/local/google/src/syzkaller-experiment-removeme/syzkaller/prog/hints_test.go:667 +0x231
testing.tRunner(0xc003303dc0, 0x15007f8)
	/usr/lib/google-golang/src/testing/testing.go:1798 +0xea
created by testing.(*T).Run in goroutine 1
	/usr/lib/google-golang/src/testing/testing.go:1857 +0x413
FAIL	github.com/google/syzkaller/prog	602.371s
FAIL
```

--- a/prog/hints_test.go
+++ b/prog/hints_test.go
@@ -655,16 +655,24 @@ func TestHintsRandom(t *testing.T) {
 	r := newRand(target, rs)
 	for i := 0; i < iters; i++ {
 		p := target.Generate(rs, 5, ct)
-		for i, c := range p.Calls {
+		for j, c := range p.Calls {
 			vals := extractValues(c)
-			for j := 0; j < 5; j++ {
+			for k := 0; k < 5; k++ {
 				vals[r.randInt64()] = true
+			}
+			// In the test mode, MutateWithHints is essentially quadratic over the number of values
+			// since we run full prog validation on each run.
+			// To avoid consuming too much time, let's just skip all calls that are too big.
+			const valsCutOff = 10000
+			if len(vals) > valsCutOff {
+				t.Logf("iter %d: skipping call %d - too big", i, j)
+				continue
 			}
 			comps := make(CompMap)
 			for v := range vals {
 				comps.Add(1, v, r.randInt64(), true)
 			}
-			p.MutateWithHints(i, comps, func(p1 *Prog) bool { return true })
+			p.MutateWithHints(j, comps, func(p1 *Prog) bool { return true })
 		}
 	}
 }

